ethexe/service/src/lib.rs

```diff
@@ -392,22 +375,44 @@ impl Service {
 
             match event {
                 Event::ServiceStarted => unreachable!("never handled here"),
+                Event::Observer(event) => match event {
```

Comment:
wdyt about writing guarantees comments here: e.g. before calling `compute.receive_synced_head` will be `// Observer service already wrote block data and events in storage` like highlighting invariants while developing. Some of this codes new parts unlikely will be changed soon

Reply:
Agree, append